packages/kit/src/runtime/server/page/render.js

```diff
@@ -361,12 +361,23 @@ export async function render_response({
 			args.push(`{\n\t\t\t\t\t\t\t${hydrate.join(',\n\t\t\t\t\t\t\t')}\n\t\t\t\t\t\t}`);
 		}
 
-		blocks.push(`Promise.all([
+		if (client.uses_env_dynamic_public && state.prerendering) {
+			blocks.push(`Promise.all([
+						import(${s(prefixed(client.start))}),
+						import(${s(prefixed(client.app))}),
+						import(${s(`${base}/${options.app_dir}/env.js`)})
```

Comment:
Does code in the client.app (or start) use the values of env during the import phase? Because that could be affected with this change.
(Thats why i load the env before everything else in my fix attempt https://github.com/bfanger/kit/tree/fix/sveltekit-v2-safari)

Reply:
`start` doesn't, but `app` _could_ because hooks are imported eagerly, and your `hooks.client.js` could import `$env/dynamic/public`.

So our solutions are essentially the same — load env, then load everything else to ensure that evaluation happens in the correct order. Because of preloading it shouldn't really make any observable difference but it's still kinda filthy.